packages/compat/src/compat-app.ts

```diff
@@ -195,12 +196,29 @@ export default class CompatApp {
       getDestinationPath: () => indexFilePath,
       annotation: 'app/index.html',
     });
-    return new this.configReplace(index, this.configTree, {
-      configPath: join('environments', `${this.legacyEmberAppInstance.env}.json`),
-      files: [indexFilePath],
-      patterns: this.configReplacePatterns,
-      annotation: 'ConfigReplace/indexTree',
-    });
+
+    return mergeTrees([
+      new this.configReplace(new ContentForSnippets(index, {}), this.configTree, {
+        configPath: join('environments', `${this.legacyEmberAppInstance.env}.json`),
+        files: [
+          // TODO: get dynamically the name of the files output by ContentForSnippets
+          'embroider-content-for-body-footer.html',
+          'embroider-content-for-head-footer.html',
+          'embroider-content-for-body.html',
+          'embroider-content-for-head.html',
```

Comment:
there are also test-head etc in tests/index.html

and you will need to use a different `env` for `tests/index.html`.
so you will also need `content-for-head` and `content-for-test-head` etc with test env

Reply:
ah, missed that one